beacon-chain/blockchain/block_processing.go

```diff
@@ -47,12 +48,12 @@ func (c *ChainService) ReceiveBlock(ctx context.Context, block *pb.BeaconBlock)
 
 	// We first verify the block's basic validity conditions.
 	if err := c.VerifyBlockValidity(block, beaconState); err != nil {
-		return nil, fmt.Errorf("block with slot %d is not ready for processing: %v", block.Slot, err)
+		return beaconState, fmt.Errorf("block with slot %d is not ready for processing: %v", block.Slot, err)
```

Comment:
We never return nil from receive block, as we should still apply fork choice regardless